src/Storages/HDFS/HDFSCommon.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <Common/config.h>
+
+#if USE_HDFS
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include <hdfs/hdfs.h>
+#include <common/types.h>
+#include <mutex>
+
+#include <Interpreters/Context.h>
+#include <Poco/Util/AbstractConfiguration.h>
+
+
+namespace DB
+{
+namespace detail
+{
+    struct HDFSFsDeleter
+    {
+        void operator()(hdfsFS fs_ptr)
+        {
+            hdfsDisconnect(fs_ptr);
+        }
+    };
+}
+
+struct HDFSFileInfo
+{
+    hdfsFileInfo * file_info;
+    int length;
+
+    HDFSFileInfo()
+        : file_info(nullptr)
+        , length(0)
+    {
+    }
+    HDFSFileInfo(const HDFSFileInfo & other) = delete;
+    HDFSFileInfo(HDFSFileInfo && other) = default;
+    HDFSFileInfo & operator=(const HDFSFileInfo & other) = delete;
+    HDFSFileInfo & operator=(HDFSFileInfo && other) = default;
+
+    ~HDFSFileInfo()
+    {
+        hdfsFreeFileInfo(file_info, length);
+    }
+};
+
+class HDFSBuilderWrapper
+{
+    hdfsBuilder * hdfs_builder;
+    String hadoop_kerberos_keytab;
+    String hadoop_kerberos_principal;
+    String hadoop_kerberos_kinit_command = "kinit";
+    String hadoop_security_kerberos_ticket_cache_path;
+
+    static std::mutex kinit_mtx;
+
+    std::vector<std::pair<String, String>> config_stor;
+
+    // hdfs builder relies on an external config data storage
+    std::pair<String, String>& keep(const String & k, const String & v)
+    {
+        return config_stor.emplace_back(std::make_pair(k, v));
+    }
+
+    void loadFromConfig(const Poco::Util::AbstractConfiguration & config, const String & config_path, bool isUser = false);
+
+    String getKinitCmd();
+
+    bool needKinit{false};
```

Comment:
Style :(

Reply:
Sorry to ask, but what is particularly wrong with style?
If this is about methods/attributes order, ok, I'll do it more regular.
The style of initialization is not unique in CH sources and I do not think that it contradicts with code style, does it?

Reply:
use snake_case for class members. 
See https://clickhouse.tech/docs/en/development/style/#names